addons/l10n_hu_edi/models/l10n_hu_edi_credentials.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+from odoo.addons.l10n_hu_edi.models.l10n_hu_edi_connection import L10nHuEdiConnectionError
+
+
+_PROTECTED_FIELDS = {'mode', 'vat', 'username', 'password', 'signature_key', 'replacement_key'}
+
+
+class L10nHuEdiCredentials(models.Model):
+    _name = 'l10n_hu_edi.credentials'
+    _description = 'Hungarian TAX Authority Login Credentials'
+    _rec_name = 'username'
+    _order = 'is_active desc'
+
+    # === User interface fields === #
+    is_active = fields.Boolean(
+        string='Active',
+        default=True,
+    )
+    is_primary = fields.Boolean(
+        string='Is Primary',
+        compute='_compute_is_primary',
+    )
+
+    # === Data fields === #
+    mode = fields.Selection(
+        selection=[
+            ('production', 'Production'),
+            ('test', 'Test'),
+        ],
+        required=True,
+        string='Server Mode',
+    )
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        string='Company',
+        required=True,
+        default=lambda self: self.env.company,
+        ondelete='cascade',
+    )
+    vat = fields.Char(
```

Comment:
Do not we have that on the company

Reply:
yes, this is a non-stored related field only for displaying on the form view (useful because the NAV credentials are linked to a particular VAT number, which needs to be the company's VAT number)